reinforcement_learning/trainer/task.py

```diff
@@ -0,0 +1,219 @@
+
+
+def build_summaries():
+    """Training and evaluation summaries."""
+    # train summaries
+    episode_reward = tf.placeholder(dtype=tf.float32, shape=[])
+    summary_vars = [episode_reward]
+    with tf.name_scope('Training'):
+        reward = tf.summary.scalar("Reward", episode_reward)
+    summary_ops = tf.summary.merge([reward])
+    # eval summary
+    eval_episode_reward = tf.placeholder(dtype=tf.float32, shape=[])
+    eval_summary_vars = [eval_episode_reward]
+    with tf.name_scope('Evaluation'):
+        eval_reward = tf.summary.scalar("EvalReward", eval_episode_reward)
+    eval_summary_ops = tf.summary.merge([eval_reward])
+
+    return summary_ops, summary_vars, eval_summary_ops, eval_summary_vars
+
+
+def log_metrics(sess, writer, summary_ops, summary_vals, metrics, test=False):
+    """Log metrics."""
+    ep_cnt, ep_r, steps, actions, noises = metrics
+    if test:
+        tf.logging.info(
+            '[TEST] Episode: {:d} | Reward: {:.2f} | AvgReward: {:.2f} | '
+            'Steps: {:d}'.format(ep_cnt, ep_r, ep_r / steps, steps))
+    else:
+        aa = np.array(actions).mean(axis=0).squeeze()
+        nn = np.array(noises).mean(axis=0).squeeze()
+        tf.logging.info(
+            '| Episode: {:d} | Reward: {:.2f} | AvgReward: {:.2f} | '
+            'Steps: {:d} | AvgAction: {} | AvgNoise: {}'.format(
+                ep_cnt, ep_r, ep_r / steps, steps, aa, nn))
+    summary_str = sess.run(summary_ops, feed_dict={summary_vals[0]: ep_r})
+    writer.add_summary(summary_str, ep_cnt)
+    writer.flush()
+
+
+def test(env, agent):
+    """Test the trained agent"""
+    tf.logging.info('Testing ...')
+    s = env.reset()
+    ep_reward = 0
+    ep_steps = 0
+    done = False
+
+    while not done:
+        if ep_steps < FLAGS.rand_steps:
+            action = agent.random_action(s)
+        else:
+            action = agent.action(s)
+        s2, r, done, info = env.step(action.squeeze().tolist())
+        ep_reward += r
+        ep_steps += 1
+        s = s2
+    return ep_reward, ep_steps
+
+
+def train():
+    """Train."""
+
+    log_dir = os.path.join(FLAGS.log_dir, 'log')
+    model_path = os.path.join(FLAGS.log_dir,
+                              'model/{}.ckpt'.format(FLAGS.agent))
+    video_dir = os.path.join(FLAGS.log_dir, 'video')
+
+    env = gym.make('BipedalWalker-v2')
+    if FLAGS.record_video:
+      ev_t = FLAGS.eval_interval
+      env = wrappers.Monitor(
+          env, video_dir,
+          video_callable=lambda ep: (ep + 1 - (ep + 1) / ev_t) % ev_t == 0)
+
+    (summary_ops, summary_vars,
+     eval_summary_ops, eval_summary_vars) = build_summaries()
+
+    with tf.Session() as sess:
+
+        if FLAGS.agent == 'DDPG':
+            agent = DDPG(env, sess, FLAGS)
+        elif FLAGS.agent == 'TD3':
+            agent = TD3(env, sess, FLAGS)
+        else:
+            raise ValueError('Unknown agent type {}'.format(FLAGS.agent))
+
+        saver = tf.train.Saver(max_to_keep=5)
```

Comment:
nit: max_to_keep could this be a flag/Constant ?

Reply:
fixed in 8cc6ed46c28df30daefc0f1ad6377dd8e00ba02c